draft-ietf-quic-transport.md

```diff
@@ -1534,15 +1534,15 @@ If 0-RTT data is accepted by the server, the server MUST NOT reduce any
 limits or alter any values that might be violated by the client with its
 0-RTT data.  In particular, a server that accepts 0-RTT data MUST NOT set
 values for the following parameters ({{transport-parameter-definitions}})
-that are smaller than the remembered value of the parameters.
-
+that are smaller than the remembered value of the parameters:
 
 * initial_max_data
 * initial_max_stream_data_bidi_local
 * initial_max_stream_data_bidi_remote
 * initial_max_stream_data_uni
 * initial_max_streams_bidi
 * initial_max_streams_uni
+* max_udp_payload_size
```

Comment:
While I agree that it is a good idea to encourage clients to remember the value, I'm not sure if we want to add max_udp_payload_size to this list. That is because doing so mandates the server to remember the max_udp_payload_size value that it advertised in the previous connection, and to not accept 0-RTT if the value has changed.

When a client reconnects from a different path, a path that the server thinks it is better to have smaller MTUs, it might be a good idea to let the server reduce the MTU. If 0-RTT packets succeed to come through (due to it being small or due to the maximum MTU being larger than what the server thinks is safe), there is no reason to drop them, just because max_udp_payload_size is going to be different.

Reply:
You are suggesting then that this can work in certain narrow cases, therefore it shouldn't be a firm requirement?

The inverse of that is that is what concerns me.  Servers accept 0-RTT with reduced limits, with reasons other than the ones you describe, and all of the 0-RTT goes into the void.  Clients will routinely attempt to fill all available 0-RTT space with data.  If the limits that the server imposes matter here.

If local constraints change in a way that the server knows about, such that it will advertise a lower limit in the transport parameter, then it is better if it doesn't accept 0-RTT.

I get that this is a "no harm" scenario in the sense that the connection won't be torn down as a result of all those packets getting lost, but the result is worse than rejecting 0-RTT because the client will mark all those packets as lost as opposed to just forgetting them.

Reply:
Well, the counter argument would be that by sending datagrams larger than 1,280 bytes, the client is already risking packet drops caused by the network. I would assume that that risk is far bigger than packet drops caused by the server reducing the size of its packet receive buffer.